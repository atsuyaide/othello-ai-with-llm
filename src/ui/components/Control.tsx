import * as style from "@app/ui/constants/style";

export interface ControlProps {
  onClickPass: () => void;
  onClickPrev: () => void;
  shouldPass: boolean;
  black: number;
  white: number;
}

export const Control = (props: ControlProps) => (
  <div style={controlStyle()} data-testid="control-component">
    <div>
      {props.black} - {props.white}
    </div>
    <div>
      <span onClick={props.onClickPrev} style={{ marginRight: 10 }}>
        Prev
      </span>
      {props.shouldPass ? (
        <span onClick={props.onClickPass}>Pass</span>
      ) : (
        <span className="disabled">Pass</span>
      )}
    </div>
  </div>
);

const controlStyle = () => ({
  width: style.mainWidth,
  height: style.controlHeight,
  margin: style.boardMargin,
});
