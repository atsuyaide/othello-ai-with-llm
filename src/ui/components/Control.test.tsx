import { fireEvent, render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { Control } from "./Control";

describe("Control", () => {
  it("renders correctly", () => {
    const onClickPrevMock = vi.fn();
    const onClickPassMock = vi.fn();
    const { container } = render(
      <Control
        black={2}
        white={3}
        onClickPrev={onClickPrevMock}
        onClickPass={onClickPassMock}
        shouldPass={true}
      />
    );

    // snapshotと比較して意図しない要素の変更がないか確認
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onClickPrev function when Prev is clicked", () => {
    const onClickPrevMock = vi.fn();
    const onClickPassMock = vi.fn();
    render(
      <Control
        black={2}
        white={3}
        onClickPrev={onClickPrevMock}
        onClickPass={onClickPassMock}
        shouldPass={true}
      />
    );

    const prevButton = screen.getByText("Prev");
    fireEvent.click(prevButton);

    expect(onClickPrevMock).toHaveBeenCalledTimes(1);
  });

  it("calls onClickPass function when Pass is clicked", () => {
    const onClickPrevMock = vi.fn();
    const onClickPassMock = vi.fn();
    render(
      <Control
        black={2}
        white={3}
        onClickPrev={onClickPrevMock}
        onClickPass={onClickPassMock}
        shouldPass={true}
      />
    );

    const passButton = screen.getByText("Pass");
    fireEvent.click(passButton);

    expect(onClickPassMock).toHaveBeenCalledTimes(1);
  });

  it("renders Pass button as disabled when shouldPass is false", () => {
    const onClickPrevMock = vi.fn();
    const onClickPassMock = vi.fn();
    render(
      <Control
        black={2}
        white={3}
        onClickPrev={onClickPrevMock}
        onClickPass={onClickPassMock}
        shouldPass={false}
      />
    );

    const passButton = screen.getByText("Pass");
    expect(passButton).toHaveClass("disabled");
  });
});
