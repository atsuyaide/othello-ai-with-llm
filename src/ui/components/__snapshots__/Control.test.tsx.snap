// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Control > renders correctly 1`] = `
<div
  data-testid="control-component"
  style="width: 356px; height: 30px; margin: 10px;"
>
  <div>
    2
     - 
    3
  </div>
  <div>
    <span
      style="margin-right: 10px;"
    >
      Prev
    </span>
    <span>
      Pass
    </span>
  </div>
</div>
`;
